packages/core/src/modules/credentials/formats/indy/IndyCredentialFormatService.ts

```diff
@@ -117,21 +124,15 @@ export class IndyCredentialFormatService extends CredentialFormatService {
     const credProposal = JsonTransformer.fromJSON(credProposalJson, CredPropose)
     await MessageValidator.validate(credProposal)
 
-    if (credProposal.credentialDefinitionId) {
-      options.credentialFormats = {
-        indy: {
-          credentialDefinitionId: credProposal?.credentialDefinitionId,
-          attributes: [],
-        },
-      }
+    options.credentialFormats.indy = {
```

Comment:
We shouldn't modify input objects like this. The only exception is the credential record